jax/_src/lru_cache.py

```diff
@@ -153,26 +169,37 @@ def _evict_if_needed(self, *, additional_size: int = 0) -> None:
     if not self.eviction_enabled:
       return
 
-    # a priority queue, each element is a tuple `(file_mtime, file, file_size)`
-    h: list[tuple[int, pathlib.Path, int]] = []
+    # a priority queue, each element is a tuple `(file_atime, key, file_size)`
+    h: list[tuple[int, str, int]] = []
     dir_size = 0
-    for file in self.path.iterdir():
-      if file.is_file() and file != self.lock_path:
-        file_size = file.stat().st_size
-        file_mtime = file.stat().st_mtime_ns
+    for cache_path in self.path.glob(f"*{_CACHE_SUFFIX}"):
+      if not cache_path.is_file():
```

Comment:
You shouldn't need this with the glob above

Reply:
Done